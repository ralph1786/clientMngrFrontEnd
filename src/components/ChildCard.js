import React from "react";
import { connect } from "react-redux";
import { deleteChild } from "../actions/childActions";
import { Link, withRouter } from "react-router-dom";
import "./ChildCard.scss";

const ChildCard = props => {
  // console.log(props.child);
  const { name, age, image, address, allergies, balance } = props.child;
  return (
    <div className="child-card animated fadeInUp">
      <h3>Name: {name}</h3>
      <p>Age: {age}</p>
      <img src={image} alt="children" />
      <address>Address: {address}</address>
      <p>Allergies: {allergies}</p>
      <p>Balance: ${balance}</p>
      <Link to="/edit">
        <button>Edit</button>
      </Link>
      <button onClick={() => props.deleteChild(props.child.id)}>Delete</button>
    </div>
  );
};

const mapDispatchToProps = dispatch => {
  return {
    deleteChild: id => dispatch(deleteChild(id))
  };
};

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(ChildCard)
);
