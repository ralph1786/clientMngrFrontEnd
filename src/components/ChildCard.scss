.child-card {
  -webkit-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  -moz-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  margin: 10px;
  padding: 5px;
  width: 400px;
  height: 550px;
  overflow: hidden;
  text-align: center;
}

.child-card img {
  width: 300px;
  height: 300px;
  border-radius: 10px;
  margin-bottom: 10px;
}

.child-card button:nth-child(odd) {
  margin: 10px;
  padding: 2px 20px;
  border-radius: 25px;
}

.child-card button:nth-child(odd):hover {
  background-color: orange;
  color: white;
  cursor: pointer;
}

.child-card button:nth-child(even) {
  margin: 10px;
  padding: 2px 20px;
  border-radius: 25px;
}

.child-card button:nth-child(even):hover {
  background-color: rgb(248, 44, 44);
  color: white;
  cursor: pointer;
}
