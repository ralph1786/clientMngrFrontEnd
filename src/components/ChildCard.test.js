import React from "react";
import ChildCard from "./ChildCard";
import { shallow } from "enzyme";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ChildCard />);
});

it("shows snapshot of ChildCard component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("checks that the child prop is not null", () => {
  expect(wrapper.props()).not.toBeNull();
});
