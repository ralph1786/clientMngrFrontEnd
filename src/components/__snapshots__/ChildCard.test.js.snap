// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows snapshot of ChildCard component 1`] = `
<Route>
  <Component />
</Route>
`;
